### PR TITLE
chore: option to disable null filter during builds

### PR DESCRIPTION
**Pull request by jacketsj**
Null filter causes slowdown right now, primarily on small datasets.

### CODE REVIEW

**Comment by wjones127** [comment]
Is the null filtering done by the `KeepFiniteVectors` transform? If so, there is an obvious performance bug we could fix: https://github.com/lancedb/lance/issues/2838

**Comment by jacketsj** [comment]
> Is the null filtering done by the `KeepFiniteVectors` transform? If so, there is an obvious performance bug we could fix: #2838

[It's currently done with actual filters.](https://github.com/lancedb/lance/blob/a1d883a28a305095ac315f8c8ed51097c21ecf34/python/python/lance/vector.py#L225) (there's two other places too).

File: python/python/lance/vector.py [hunk 1]
```diff
@@ -201,6 +201,7 @@ def train_ivf_centroids_on_accelerator(
     *,
     sample_rate: int = 256,
     max_iters: int = 50,
+    disable_null_filter: bool = False,
```
Comment (chebbyChefNEQ) [inline python/python/lance/vector.py:204]: maybe call this `check_nan`? nulls here are really nans or nulls

File: python/python/lance/dataset.py [hunk 2]
```diff
@@ -1503,6 +1504,10 @@ def create_index(
         storage_options : optional, dict
             Extra options that make sense for a particular storage connection. This is
             used to store connection parameters like credentials, endpoint, etc.
+        disable_null_filter: bool
+            Defaults to false. True is UNSAFE. Disable the null filter used for nullable
+            columns. Will crash if any nulls are present in the column. Obtains a small
+            speed boost.
```
Comment (chebbyChefNEQ) [inline python/python/lance/dataset.py:1510]: maybe we should note the condition under which `True` is safe?

**Review by chebbyChefNEQ — APPROVED** [review]

**Pull request merged**